qiskit/quantum_info/primitives/utils.py

```diff
@@ -0,0 +1,124 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Utility functions for primitives
+"""
+
+from __future__ import annotations
+
+import sys
+
+from qiskit.circuit import ParameterExpression, QuantumCircuit
+from qiskit.extensions.quantum_initializer.initializer import Initialize
+from qiskit.quantum_info.operators import SparsePauliOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.quantum_info.states import Statevector
+
+if sys.version_info >= (3, 8):
+    # pylint: disable=no-name-in-module
+    from typing import Protocol, runtime_checkable
+else:
+    from typing_extensions import Protocol, runtime_checkable
+
+
+@runtime_checkable
+class PauliSumOp(Protocol):
+    """Protocol for opflow.PauliSumOp."""
```

Comment:
what is this for?

Reply:
I didn't want to make quantum_info dependent on opflow, so I implemented a Protocol to achieve this. If we had made it dependent, the code would have been complicated by the cyclic dependency.